src/core/xref.js

```diff
@@ -143,6 +128,51 @@ class XRef {
       }
       warn(`XRef.parse - Invalid "Root" reference: "${ex}".`);
     }
+
+    if (encrypt instanceof Dict) {
+      const ids = trailerDict.get("ID");
+      const fileId = ids?.length ? ids[0] : "";
+
+      const stmF = encrypt.get("StmF")?.name ?? "Identity";
+      const strF = encrypt.get("StrF")?.name ?? "Identity";
+      const eff = encrypt.has("EFF") ? encrypt.get("EFF")?.name : strF;
+      const cryptFilters = encrypt.get("CF") || Dict.empty;
+      // Check if only the file attachments are encrypted.
+      if (
+        stmF === "Identity" &&
+        strF === "Identity" &&
+        eff !== "Identity" &&
+        cryptFilters.get(eff)?.get("CFM") !== "None"
+      ) {
+        let hasEncryptedAttachments = false;
+        if (root instanceof Dict) {
+          const names = root.get("Names");
+          if (names instanceof Dict && names.has("EmbeddedFiles")) {
+            const nameTree = new NameTree(names.getRaw("EmbeddedFiles"), this);
+            const attachments = nameTree.getAll();
```

Comment:
Not sure about how much performance matters in this case, but we could have a `nameTree.isEmpty()` method to avoid having to extract all the entries just to check if any exists.